src/auth/dto/login-credentials.dto.ts

```diff
@@ -0,0 +1,12 @@
+import { IsString, MinLength, MaxLength, Matches } from "class-validator";
+
+export class LoginCredentialsDTO {
+
+    public email: string;
+
+    public phonenumber: string;
+
+    public username: string;
```

Comment:
We agreed that we will have one field so FE will send you one property (which will contain one of these three) and BE should check which one the property refer to from DB or maybe by validation